polynote-server/src/main/scala/polynote/server/repository/fs/package.scala

```diff
@@ -0,0 +1,38 @@
+package polynote.server.repository
+
+import java.nio.file.{FileSystemNotFoundException, Path}
+
+import polynote.kernel.{BaseEnv, GlobalEnv}
+import zio.{Has, RIO, ULayer, URIO, ZIO, ZLayer}
+
+package object fs {
+
+  type FileSystems = Has[FileSystems.Service]
+
+  object FileSystems {
+
+    trait Service {
```

Comment:
The idea with this is to provide filesystem access in the environment. Then a mock filesystem can be used in testing quite easily. We already had this abstraction which does just path-based read/write/list/delete operations, so I just added a service around it.

`forScheme` is just planning for when remote filesystems become a thing... it's not used anywhere (probably YAGNI violation, oh well)